src/ibex_bluesky_core/plans/reflectometry/autoalign_utils.py

```diff
@@ -0,0 +1,232 @@
+            and the beam. e.g Gaussian.
+        periods (bool): Are DAE periods being used. Defaults to True.
+        save_run (bool): Should runs be saved. Defaults to True.
+        is_good_fit (Callable[[ModelResult, float], bool] | None): User provided checks on the
+            optimised value, must return True if result is sensible.
+        problem_found_plan (Callable[[], Generator[Msg, None, None]] | None]):
+            A callback for what to do if the optimised value is not found to be sensible.
+
+    Returns:
+        Whether there was a problem during runtime. Returns True if there was. (bool).
+
+    """
+    print(f"Scanning over {alignment_param.name} with a relative scan range of {rel_scan_range}.")
+
+    init_mot_pos: float = yield from bps.rd(alignment_param)  # type: ignore
+
+    def _inner() -> Generator[Msg, None, ISISCallbacks]:
+        icc = yield from scan(
+            dae=dae,
+            block=alignment_param,
+            start=-rel_scan_range / 2,
+            stop=rel_scan_range / 2,
+            num=num_points,
+            model=fit_method,
+            periods=periods,
+            save_run=save_run,
+            rel=True,
+        )
+
+        return icc
+
+    icc = yield from _inner()
+
+    alignment_param_value = _get_alignment_param_value(icc, fit_param)
+
+    if not _check_parameter(
+        alignment_param_value=alignment_param_value,
+        result=icc.live_fit.result,
+        init_mot_pos=init_mot_pos,
+        rel_scan_range=rel_scan_range,
+        is_good_fit=is_good_fit,
+    ):
+        print(
+            f"""This failed one or more of the checks on {alignment_param.name}"""
+            f""" with a scan range of {rel_scan_range}."""
+        )
+        print(f"Value found for {fit_param} was {alignment_param_value}.")
+
+        yield from problem_found_plan()
+
+        def inp() -> Generator[Msg, None, str]:
+            choice = yield from call_sync(
+                input,
+                """Type '1' if you would like to re-scan or type '2' to re-zero"""
+                f""" at {alignment_param_value} and keep going.""",
+            )
+
+            if choice not in {"1", "2"}:
+                choice = yield from inp()
+
+            return choice
+
+        choice = yield from inp()
+        if choice == "1":
+            return (icc, True)
+
+    print(f"Moving {alignment_param.name} to {alignment_param_value}.")
+    yield from bps.mv(alignment_param, alignment_param_value)  # type: ignore
+
+    return (icc, False)
+
+
+class OptimiseAxisParams(TypedDict):
+    """Type hints for optimise_axis_against_intensity."""
+
+    num_points: NotRequired[int]
+    periods: NotRequired[bool]
+    save_run: NotRequired[bool]
+    problem_found_plan: NotRequired[Callable[[], Generator[Msg, None, None]]]
+    is_good_fit: NotRequired[Callable[[ModelResult, float], bool] | None]
+
+
+def optimise_axis_against_intensity(  # noqa: D417
```

Comment:
probably need to say in the docs that this moves the movable to the optimum position, so you _can_ redefine it in your top level plan if you want to